### PR TITLE
Make FlutterCommand.startProcess() return details to handle specific issues in clients.

### PR DESCRIPTION
**Pull request by scheglov**
This fixes https://github.com/flutter/flutter-intellij/issues/1622 by showing a meaningful error instead of `@NotNull` failure.

image

### CODE REVIEW

File: src/io/flutter/sdk/FlutterCommandStartResult.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  @NotNull
+  public final FlutterCommandStartResultStatus status;
+
+  @Nullable
+  public final OSProcessHandler processHandler;
+
+  @Nullable
+  public final ExecutionException exception;
+
+  public FlutterCommandStartResult(@NotNull FlutterCommandStartResultStatus status,
```
Comment (devoncarew) [inline src/io/flutter/sdk/FlutterCommandStartResult.java]: I'd add a few more of these constructors - I note that most call sites pass in null for the last param.

File: src/io/flutter/run/test/TestLaunchState.java [hunk 2]
```diff
@@ -91,7 +93,19 @@ static TestLaunchState create(@NotNull ExecutionEnvironment env, @NotNull TestCo
...
   protected ProcessHandler startProcess() throws ExecutionException {
     final RunMode mode = RunMode.fromEnv(getEnvironment());
-    return fields.run(getEnvironment().getProject(), mode);
+    final FlutterCommandStartResult result = fields.run(getEnvironment().getProject(), mode);
+    switch (result.status) {
+      case OK:
+        assert result.processHandler != null;
+        return result.processHandler;
+      case ANOTHER_RUNNING:
+        throw new ExecutionException("There is another Flutter command running");
```
Comment (devoncarew) [inline src/io/flutter/run/test/TestLaunchState.java]: ==> `another Flutter instance already running`?

File: src/io/flutter/sdk/FlutterCommandStartResult.java (hunk 1 above)
Comment (scheglov) [reply src/io/flutter/sdk/FlutterCommandStartResult.java]: Done.

File: src/io/flutter/run/test/TestLaunchState.java (hunk 2 above)
Comment (scheglov) [reply src/io/flutter/run/test/TestLaunchState.java]: Done.

**Pull request merged**